build.gradle

```diff
@@ -150,80 +153,85 @@ tasks.withType(SpotBugsTask) {
 }
 
 codenarc {
-  toolVersion '1.5'
-  sourceSets = [sourceSets.test]
+    toolVersion '1.5'
+    sourceSets = [sourceSets.test]
 }
 
 test {
-  systemProperties['hudson.model.DownloadService.noSignatureCheck'] = 'true'
-  // override default timeout
-  systemProperties['jenkins.test.timeout'] = 300
-  ignoreFailures = gradle.ciJenkinsBuild
-  maxParallelForks = project.maxParallelForks
-  retry {
-    if (gradle.isCi) {
-      maxRetries = 1
-      maxFailures = 10
+    systemProperties['hudson.model.DownloadService.noSignatureCheck'] = 'true'
+    // override default timeout
+    systemProperties['jenkins.test.timeout'] = 300
+    ignoreFailures = gradle.ciJenkinsBuild
+    maxParallelForks = project.maxParallelForks
+    retry {
+        if (gradle.isCi) {
+            maxRetries = 1
+            maxFailures = 10
+        }
     }
-  }
 }
 
 def checkArchiveManifest(File archive) {
-  new ZipFile(archive).withCloseable { archiveZip ->
-    archiveZip.getInputStream(archiveZip.getEntry('META-INF/MANIFEST.MF')).withStream {
-      assert it.text.contains("Plugin-Version: ${project.version}"): "Wrong metadate in file ${archive} - run a clean build"
+    new ZipFile(archive).withCloseable { archiveZip ->
+        archiveZip.getInputStream(archiveZip.getEntry('META-INF/MANIFEST.MF')).withStream {
+            assert it.text.contains("Plugin-Version: ${project.version}"): "Wrong metadate in file ${archive} - run a clean build"
+        }
     }
-  }
 }
 
 tasks.withType(AbstractArchiveTask) {
-  inputs.property('pluginVersion') {
-    project.version
-  }
+    inputs.property('pluginVersion') {
+        project.version
+    }
+}
+
+// See https://github.com/jenkinsci/gradle-jpi-plugin/blob/master/src/main/kotlin/org/jenkinsci/gradle/plugins/manifest/GenerateJenkinsManifestTask.kt#L69
+tasks.withType(GenerateJenkinsManifestTask).configureEach {
+    version = project.version.toString().replace("-SNAPSHOT", "-dev")
```

Comment:
🤔 I am not sure you always want to do this or only when running tests. I suppose when you upload, you don't want to use a fixed `-dev` version even for snapshot builds.

Reply:
Yes, in general, we don't want `-dev` to leak anywhere. At the moment we don't have use cases except testing where we use `-SNAPSHOT`. It's a temporary "hack". Hopefully, my PR: https://github.com/jenkinsci/gradle-jpi-plugin/pull/210 will be accepted and we will be able to get rid of it soon.

Reply:
This may have an effect for manual testing when you deploy the HPI file to your locally running Jenkins, right?